SYCL/Regression/device_num.cpp

```diff
@@ -19,12 +19,10 @@ using namespace std;
 const std::map<info::device_type, std::string> DeviceTypeStringMap = {
     {info::device_type::cpu, "cpu"},
     {info::device_type::gpu, "gpu"},
-    {info::device_type::host, "host"},
```

Comment:
Do I understand it correctly, that this test fails in the PR, because we still return host device in our implementation?

```
[2022-09-02T12:58:16.813Z] --

[2022-09-02T12:58:16.813Z] Exit Code: -6

[2022-09-02T12:58:16.813Z] 

[2022-09-02T12:58:16.813Z] Command Output (stdout):

[2022-09-02T12:58:16.813Z] --

[2022-09-02T12:58:16.813Z] $ ":" "RUN: at line 1"

[2022-09-02T12:58:16.813Z] note: command had no output on stdout or stderr

[2022-09-02T12:58:16.813Z] $ "/netbatch/donb2930500_00/runDir/jenkins-dir/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CI-Linux/llvm.obj/bin/clang++" "-fsycl" "-fsycl-targets=spir64" "/netbatch/donb2930500_00/runDir/jenkins-dir/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CI-Linux/llvm-test-suite/SYCL/Regression/device_num.cpp" "-o" "/netbatch/donb2930500_00/runDir/jenkins-dir/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CI-Linux/build/SYCL/Regression/Output/device_num.cpp.tmp.out"

[2022-09-02T12:58:16.813Z] note: command had no output on stdout or stderr

[2022-09-02T12:58:16.813Z] $ ":" "RUN: at line 2"

[2022-09-02T12:58:16.813Z] note: command had no output on stdout or stderr

[2022-09-02T12:58:16.813Z] $ "env" "PRINT_FULL_DEVICE_INFO=1" "/netbatch/donb2930500_00/runDir/jenkins-dir/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CI-Linux/build/SYCL/Regression/Output/device_num.cpp.tmp.out"

[2022-09-02T12:58:16.813Z] note: command had no output on stdout or stderr

[2022-09-02T12:58:16.813Z] $ ":" "RUN: at line 3"

[2022-09-02T12:58:16.813Z] note: command had no output on stdout or stderr

[2022-09-02T12:58:16.813Z] $ "env" "SYCL_DEVICE_FILTER=0" "env" "TEST_DEV_CONFIG_FILE_NAME=/netbatch/donb2930500_00/runDir/jenkins-dir/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CI-Linux/build/SYCL/Regression/Output/device_num.cpp.tmp1.conf" "/netbatch/donb2930500_00/runDir/jenkins-dir/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CI-Linux/build/SYCL/Regression/Output/device_num.cpp.tmp.out"

[2022-09-02T12:58:16.813Z] # command stderr:

[2022-09-02T12:58:16.813Z] device_num.cpp.tmp.out: /netbatch/donb2930500_00/runDir/jenkins-dir/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CI-Linux/llvm-test-suite/SYCL/Regression/device_num.cpp:54: DevInfoEntry getDeviceInfoByName(const std::string &, const std::map<DevInfoEntry, std::string> &) [DevInfoEntry = sycl::info::device_type]: Assertion `it != mapWithNames.end()' failed.

[2022-09-02T12:58:16.813Z] 

[2022-09-02T12:58:16.813Z] error: command failed with exit status: -6

[2022-09-02T12:58:16.813Z]
```

I would suggest to leave those maps as-is for now, because we only deprecate those enumerations, but not remove them. It will allow to merge the test right away without introducing a new failure

Reply:
Yes, that does seem to be the case. I have reintroduced them for now.